### PR TITLE
feat(llm): added the process of text2gql in graphrag V1.0

### PR DESCRIPTION
**Pull request by vichayturen**
address #10 

1. added the process of intelligent generated gremlin retrivecal
2. added text2gremlin block in rag app

### CODE REVIEW

**Comment by simon824** [comment]
After implementing `text2gremlin` here, can [gremlin_generate_web_demo.py ](https://github.com/apache/incubator-hugegraph-ai/blob/main/hugegraph-llm/src/hugegraph_llm/demo/gremlin_generate_web_demo.py) be removed?

**Comment by vichayturen** [comment]
> After implementing `text2gremlin` here, can [gremlin_generate_web_demo.py ](https://github.com/apache/incubator-hugegraph-ai/blob/main/hugegraph-llm/src/hugegraph_llm/demo/gremlin_generate_web_demo.py) be removed? 

Yes, and it has been removed in new commit.

**Comment by imbajin** [comment]
We should add a `flag` value under the interface of graph query rag/graph: 
- `1` represents text2gql accurate matching success
- `0` represents generalization matching success
- `-1` represents no relevant information

File: hugegraph-llm/src/hugegraph_llm/operators/index_op/gremlin_example_index_query.py [hunk 1]
```diff
@@ -19,21 +19,43 @@
...
+        query = context.get("query")
+        assert query, "query is required"
+        if self.num_examples == 0:
+            context["match_result"] = []
+        else:
+            if "query_embedding" in context and isinstance(context["query_embedding"], list):
+                query_embedding = context["query_embedding"]
+            else:
+                query_embedding = self.embedding.get_text_embedding(query)
+            context["match_result"] = self.vector_index.search(query_embedding, self.num_examples, dis_threshold=2)
```
Comment (imbajin) [inline hugegraph-llm/src/hugegraph_llm/operators/index_op/gremlin_example_index_query.py:52]: why we set `dis_threshold` to 2? (means always match topK?)

Seems not reasonable

**Review by imbajin — APPROVED** [review]
Merg it & split it to separate PR for dev

**Pull request merged**